### PR TITLE
Adding first draft of [OTEL Kubernetes] Cluster Overview Dashboard

### PR DESCRIPTION
**Pull request by gizas**
- Enhancement

## Proposed commit message

Adding a new `[OTEL Kubernetes] Cluster Overview` dashboard in k8s integration

Please explain:

- WHAT: A new dashboard that visualises the needed metrics for K8s Cluster observability
- WHY:  To support the user experience of Kubernetes Observability users that use OTEL collector 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

1. Create a kind cluster `kind create cluster`
2. Clone this repo
3. Navigate to `elastic/integrations/packages/kubernetes`
4. Build package by running `elastic-package build`
5. Install the elastic-agent-otel distribution from https://github.com/elastic/k8scollector/blob/otel/helm/elastic-agent/elastic-otel-collector_all.yaml. Run `kubectl apply -f elastic-otel-collector_all.yaml`
6. Spin up a new `elastic-package stack up -d -v --version=8.15.0-SNAPSHOT`
7. The new Kubernetes 1.64.0 integration should be available. Install it
8. Navigate to Dashboards and open `[OTEL Kubernetes] Cluster Overview `

## Screenshots

Screenshot 2024-07-10 at 1 35 47 PM

### CODE REVIEW

File: packages/kubernetes/kibana/dashboard/kubernetes-e31db833-de44-4f74-a2df-f7c1337bb1be.json [hunk 1]
```diff
@@ -436,85 +340,1819 @@
...
+                                            "columnOrder": [
+                                                "7113c7e7-1af9-4350-b5d2-57abcb60c633",
+                                                "af01f323-afc0-4b55-b453-8da15facfc28",
+                                                "68bd6f9e-6894-4991-b2ff-fac3a4461b2b"
+                                            ],
+                                            "columns": {
+                                                "68bd6f9e-6894-4991-b2ff-fac3a4461b2b": {
+                                                    "dataType": "number",
+                                                    "isBucketed": false,
+                                                    "label": "Average of k8s.node.cpu.utilization",
```
Comment (ChrsMark) [inline packages/kubernetes/kibana/dashboard/kubernetes-e31db833-de44-4f74-a2df-f7c1337bb1be.json:1449]: The way it is reported right now by the Collector is not correct. I'd suggest keeping this out for now or use the `k8s.node.cpu.usage` instead.

You can find more info at:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27885#issuecomment-2225039360
- https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6a5b9e673a433ad2c6ed7ef2526a8920fd18cf0d/receiver/kubeletstatsreceiver/metadata.yaml#L96

There is a fix for that at https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/e2483538dde00ad4de2462ff6b22979557880a1f but i'm not sure yet if we will first land this or if we should first deprecate and remove the metrics and then add the correct metric back.
You can follow https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27885 for updates.

**Comment by gizas** [comment]
Adding another iteration

- Updated Node visualisation with k8s.node.cpu.usage
- Added tables for memory/ cpu utilisation. Those have coloured cells. Although there seems a problem with colouring. See memory in screenshot, it has red colour although value is 15%
- Added network errors 
- Added node filesystem usage in the bottom

Screenshot 2024-07-18 at 5 29 51 PM

File: packages/kubernetes/kibana/dashboard/kubernetes-e31db833-de44-4f74-a2df-f7c1337bb1be.json [hunk 2]
```diff
@@ -0,0 +1,5986 @@
...
+                    "y": 26
+                },
+                "panelIndex": "66b0021b-9111-452e-9156-be85ab159e3e",
+                "title": "Top CPU Pods - Node Percentage (Average)",
+                "type": "lens"
+            },
+            {
+                "embeddableConfig": {
+                    "attributes": {
+                        "description": "Average of 100 Top Pod CPU Usage based on kubernetes.pod.usage.limit.pct",
```
Comment (ChrsMark) [inline packages/kubernetes/kibana/dashboard/kubernetes-e31db833-de44-4f74-a2df-f7c1337bb1be.json:3401]: If those dashboards aim OTel users the namings should meet the [semantic convention expectations](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#instrument-naming), using the `Utilization` or `Usage` according to the underlying metric's name/notion.

**Review by ChrsMark — COMMENTED** [review]
I might miss some content here but something that looks unclear is which type of data this Dashboard aims to consume.

Is this going to use translated metrics coming from the Infra Metrics Processor along with OTel native metrics?

Since this is focused on OTel users and the translation lib emits both the translated and the OTel native ones, why not to use the native ones here directly?

So instead of `kubernetes.pod.cpu.usage.limit.pct` to use the `k8s.pod.cpu_limit_utilization` directly?

**Comment by gizas** [comment]
Closed as work was done in different PR

**Pull request closed without merging**